src/pages/profile/index.ts

```diff
@@ -165,35 +185,150 @@ const profileItems = [
+            })
+        ],
+        button: new Button({
+            text: 'СОХРАНИТЬ',
+            attr: {
+                class: 'button'
+            },
+            events: {
+                click: (e:Event) => {
+                    const inputs: NodeList | null = document.querySelectorAll('.input__item_popup');
+                    let data = {
+                        oldPassword: '',
+                        newPassword: ''
+                    };
+                    inputs.forEach(input => {
+                        const inputName = (input as HTMLInputElement).name;
+                        if ((input as HTMLInputElement).value) {
+                            data[inputName] = (input as HTMLInputElement).value;
+                            (input as HTMLInputElement).value = '';
+                        }
+                    });
+                    UserController.changePassword(data);
+                }
+            } 
+        }),
+        attr: {
+            class: 'popup'
+        },
+        events: {
+            click: (e: Event) => {
+                const t = e.target;
+                if (t && (t as HTMLElement).className === 'popup__close') {
+                    const popup = document.querySelector('.popup');
+                    popup?.setAttribute('style', 'display: none;');
+                }
+            }
+        }
+    }),
     buttons: [
         new Button({
-            text: 'СОХРАНИТЬ',
+            text: 'СМЕНИТЬ ПАРОЛЬ',
             attr: {
                 class: 'button button_profile'
             },
             events: {
                 click: (e: Event) => {
+                    const popup: HTMLDivElement | null = document.querySelector('.popup');
+                    popup?.setAttribute('style', 'display: block');
+                }
+            }
+        }),
+        new Button({
+            text: 'СОХРАНИТЬ',
+            attr: {
+                class: 'button button_profile'
+            },
+            events: {
+                click: async (e: Event) => {
                     e.preventDefault();
                     const inputs = document.querySelectorAll('input');
-                    const userData = {};
+                    const userData: any = {};
                     inputs.forEach(input => {
                         const inputName = input.name;
-                        userData[inputName] = input.value;
-                        input.value = '';
+                        if (input.value) {
+                            userData[inputName] = input.value;
+                            input.value = '';
+                        } else {
+                            userData[inputName] = input.placeholder;
+                        }
                     });
-                    console.log(userData);
+                    userData.display_name = `${userData.first_name} ${userData.second_name}`
+                    UserController.changeProfile(JSON.stringify(userData) as any);
```

Comment:
метод асинхронный, должен вызываться через await 
```suggestion
                    await UserController.changeProfile(JSON.stringify(userData) as any);
```